src/runtime/containerd-shim-v2/start.go

```diff
@@ -37,8 +37,6 @@ func startContainer(ctx context.Context, s *service, c *container) error {
 		}
 		go watchSandbox(ctx, s)
 
-		// We don't rely on the context passed to startContainer as it can be cancelled after
-		// this rpc call.
 		go watchOOMEvents(ctx, s)
```

Comment:
I realized I changed this from `s.ctx` to `ctx`, but didn't make a change in `watchOOMEvents()` where I should have. Since it checks for `case <-ctx.Done()` which originally was with `s.ctx` passed to it, it actually should be checking if `s.ctx` is canceled. The service struct is already an argument in that call so it is a quick change. Also, in `watchOOMEvents()`, the context in `GetOOMEvent(ctx)` should stay the same since it is the local context used for tracing. Then the comment could be something like "We use s.ctx to check for cancellation of the shim context and the context passed to startContainer for tracing."

Reply:
Hi @cmaf, do you mean only replace the comment at Line 40-41? 

If so,  how about this:
```
We use s.ctx(`ctx` derived from `s.ctx`) to check for cancellation of the shim context and the context passed to startContainer for tracing.
```
I added **`ctx` derived from `s.ctx`** for  describe the relation of `ctx` and `s.ctx`

Reply:
@liubin, yeah that looks good. Could you also make this change where the check happens?

```
diff --git a/src/runtime/containerd-shim-v2/wait.go b/src/runtime/containerd-shim-v2/wait.go
index ad2a3af8..81a05aa2 100644
--- a/src/runtime/containerd-shim-v2/wait.go
+++ b/src/runtime/containerd-shim-v2/wait.go
@@ -142,7 +142,7 @@ func watchOOMEvents(ctx context.Context, s *service) {
 
 	for {
 		select {
-		case <-ctx.Done():
+		case <-s.ctx.Done():
 			return
 		default:
 			containerID, err := s.sandbox.GetOOMEvent(ctx)
```

Reply:
Hi @cmaf I have updated it as your advise.

Reply:
@liubin, I'd prefer this in its own commit, as it deserves a backport to the `stable-2.1` branch (while the comment changes can live only as part of `main`).

Reply:
Ok, I will create a new PR to only fix this. @fidencio